sunpy/net/hek/hek.py

```diff
@@ -96,6 +106,95 @@ def _parse_times(table):
                 table[tkey].format = 'iso'
         return table
 
+    @staticmethod
+    def _parse_unit(table, attribute, is_coord_prop = False):
+        unit_attr=""
+        if is_coord_prop:
+            unit_attr = "event_coordunit"
+        else:
+            unit_attr = attribute["unit_prop"]
+        for row in table:
+            if unit_attr in table.colnames and row[unit_attr] not in ["", None] and table[attribute["name"]].unit is not None:
+                table[attribute["name"]].unit = HEKClient._get_unit(attribute, row[unit_attr], is_coord_prop= is_coord_prop)
+                break
+        return table
+
+    @staticmethod
+    def _parse_astropy_unit(str):
+        try:
+            unit = u.Unit(str)
+        except ValueError:
+            try:
+                unit = u.Unit(str.lower())
+            except ValueError:
+                unit = u.Unit(str.capitalize())
```

Comment:
What happens if this fails?

Reply:
I have changed this function a bit. Could you please check and let me know if it's fine?